Add a cryosphere default config file

xylar opened the pull request:
This file `configs/polarRegions.conf` contains new "defaults" for E3SM cryosphere runs that should override the defaults in `mpas_analysis/config.default`

Also updates values in `config.default` for colormap indices and colorbar ranges for the MOC, Hovmoller temperature and salinity anomaly, and Argo bias plots.

Example usage (see `configs/README.md`):
```
python -m mpas_analysis configs/polarRegions.conf my_favorite.conf --generate=climatologyMapSSH --purge
```

If using the conda package, you'll need to download https://raw.githubusercontent.com/MPAS-Dev/MPAS-Analysis/develop/configs/polarRegions.conf and then run something like:
```
mpas_analysis polarRegions.conf my_favorite.conf --generate=climatologyMapSSH --purge
```

xylar commented:
@milenaveneziani, please add a commit to this branch with your suggested modifications.  After that, we can discuss further here.

xylar commented on configs/cryosphere.conf:
```diff
@@ -0,0 +1,33 @@
+## and WOA18 climatological data
+
+# A dictionary with keywords for the norm
+normArgsResult = {'vmin': -2., 'vmax': 2.}
```
This range is intended to be consistent with SOSE analysis tasks.

xylar commented on configs/cryosphere.conf:
```diff
@@ -0,0 +1,33 @@
+## and WOA18 climatological data
+
+# A dictionary with keywords for the norm
+normArgsResult = {'vmin': 33.8, 'vmax': 35.0}
```
Again, for consistency with SOSE.

xylar commented:
To get the conversation started, Here are 2 example from @milenaveneziani's analysis (top) and a similar plot form @stephenprice's (bottom) with default settings:
Sea surface:
![image](https://user-images.githubusercontent.com/4179064/67470262-bfa9c600-f64d-11e9-84a8-119576013d81.png)
![image](https://user-images.githubusercontent.com/4179064/67470247-b7ea2180-f64d-11e9-900e-8d3c0611abe8.png)

Sea floor:
![image](https://user-images.githubusercontent.com/4179064/67470032-66419700-f64d-11e9-8a4e-66dc9163284f.png)
![image](https://user-images.githubusercontent.com/4179064/67469997-588c1180-f64d-11e9-8853-74534d1830c1.png)

@milenaveneziani's range seems good for the surface but our community expects the default range for the sea floor.  Currently, MPAS-Analysis doesn't support different ranges at different depths and I think it would be confusing and clumsy to specify.  One option would be a different range for the surface than for all other depths.

xylar commented:
Similarly, @milenaveneziani's salinity range is appropriate for the surface but contrast is lost at depth.

xylar commented:
@milenaveneziani, I wonder if your MOC and Argo changes should just become the new defaults.

xylar commented:
@milenaveneziani, you had reduced the vertical resolution of the transects from 10 m to 30 m.  Was that just to make things run faster?  I'd rather keep the 10-m resolution unless there's a problem with it.

milenaveneziani commented:
I wish I could reply to each single comment box..

Anyways, good idea to have a config file for cryo stuff. Maybe we should call it polarRegions.conf, or something like that?

Re vertical sections: 1) yes, I have been wanting to modify the section at 288 myself for a while; and 2) discrete vs continuous colormaps: indeed we have disagreed in the past and I don't mean to push my preference. I personally like to see an indication of discrete stratification in vertical sections, and that is why I changed it only for vertical sections in my own config file.

Re changes for Hovmoeller diagrams, MOC and Argo: yes, I can add a commit on config.default for those.

Re vertical resolution of the transects from 30 to 10m: I actually do not remember why I did that.. Maybe it was related to speed, but I honestly can't remember.

milenaveneziani commented:
> 1) yes, I have been wanting to modify the section at 288 myself for a while

wait, now I remember why I didn't do this before: we can only set one single pair of latmin,latmax for all SOSE sections.

xylar commented:
> Maybe we should call it polarRegions.conf, or something like that?

Isn't `cryosphere` pretty much the same thing as `polarRegions`?

xylar commented:
>> yes, I have been wanting to modify the section at 288 myself for a while
> wait, now I remember why I didn't do this before: we can only set one single pair of latmin, latmax for all SOSE sections.

Right, that's intentional because it's just too complicated to do anything else.  So you'd need to move that transect to the west to put it just past Ronne.  Or we just live with the weird artifact.

milenaveneziani commented:
> So you'd need to move that transect to the west to put it just past Ronne
good idea.

I'm fine with the name of the config file. I guess I always think of cryo as more related to ice, but that's probably just me.

One other thing I just noticed: maybe we don't need to add seaice freshwater flux for the time series as default. Looking at it by itself is not very informative. (I had in my config file when I was testing the latest PR on time series, and then forgot to remove it)

xylar commented:
> I'm fine with the name of the config file. I guess I always think of cryo as more related to ice, but that's probably just me.

Okay, fair enough.  I'm good with polarRegions.conf, then.

> One other thing I just noticed: maybe we don't need to add seaice freshwater flux for the time series as default. Looking at it by itself is not very informative. (I had in my config file when I was testing the latest PR on time series, and then forgot to remove it)

Ah, I see.  I'll remove that commit, then.

milenaveneziani commented:
I take that back about changing the section at 288 (72W): I had chosen that longitude to make it possible to compare the section with Fig. 10 of the Whithworth paper (in fact, all the other sections I added were chosen for the same purpose). I'd leave it like that and just live with the weird plot, if that's OK with the others.

milenaveneziani commented:
> Is this the Whitworth paper you're talking about?
https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1029/AR075p0001

yup. In particular, Figs. 2-10. We could document it: where do you think that would be the best place to do it?

milenaveneziani commented:
maybe in this config file.

xylar commented on configs/polarRegions.conf:
```diff
@@ -0,0 +1,143 @@
+# Amudsen, Thwaites (253=107W), Bellingshausen (280=80W),
+# West Antarctica (288=72W)
+# Note: Transects at XXX, XXX, XXX, XXX and XXX are chosen for comparison with
+#       Whitworth et al. (XXXX) doi: XXXX/XXXX
```
@milenaveneziani, could you edit this to have the right info here?  I think you can just edit the file directly on GitHub if you want to.  You can even use the "insert a suggestion" right here but I think it only edits one line at a time.

milenaveneziani replied:
really? I didn't know I could do that. Let me  try it.

xylar commented on configs/polarRegions.conf:
```diff
@@ -0,0 +1,143 @@
+# Amudsen, Thwaites (253=107W), Bellingshausen (280=80W),
+# West Antarctica (288=72W)
+# Note: Transects at 325, 145, 160, 184, 198 and 288 are chosen for comparison with
+#       Whitworth et al. (1998) doi: 10.1029/AR075p0001, Figs. 3, 7, 6, 5, 4, 10, respectively.
```
Great, thanks!

xylar commented:
Eh, I'm good with the discrete.  You've put a lot of thinking into it.

xylar commented:
Rebased after the merge of #643.  I also updated the example config files, the associated README and the job scripts on cori to reflect this new config file.

xylar commented:
@matthewhoffman, @stephenprice and @darincomeau, let me know if you are interested in looking at this and reviewing the PR.

darincomeau commented:
@xylar I don't have strong feelings here. That said, after a quick look through these changes look good to me. Would you prefer I approve, or remove myself as reviewer?

xylar commented:
@matthewhoffman and @stephenprice, I know things have been busy with Jan's visit and all.  I'm taking you off as reviewers and I'm going to merge.  But if you have feedback on what you think the default settings should be, we can update them later.
